### ISSUE TITLE
aws_lambda_permission: ResourceNotFoundException: Function not found

### ISSUE THREAD

**Issue by zentavr** [terraform-providers/terraform-provider-aws#7214]
### Terraform Version

```
Terraform v0.11.11
+ provider.aws v1.55.0
```

### Affected Resource(s)

* aws_lambda_permission

### Terraform Configuration Files

```hcl
resource "aws_lambda_function" "lambda" {
...
...
}

resource "aws_alb_target_group" "events" {
  name = "${var.environment}-events-lambda"
  target_type = "lambda"

  tags {
    Name = "${var.environment}-events-lambda"
    Description = "ALB Target Group for events calls in ${var.environment} application environment"
    Environment = "${var.environment}"
    Origin = "terraform"
  }
}

resource "aws_lambda_permission" "events_endpoint_lambda_alb_permission" {
  statement_id = "Allow-events-alb-invoke"
  action = "lambda:InvokeFunction"
  function_name = "${aws_lambda_function.lambda.name}"
  principal = "elasticloadbalancing.amazonaws.com"
  source_arn = "${aws_alb_target_group.events.arn}"
}
```

### Debug Output

```
aws_lambda_function.lambda: Creation complete after 3s (ID: dev_events_endpoint)

Error: Error applying plan:

1 error(s) occurred:

* aws_lambda_permission.events_endpoint_lambda_alb_permission: 1 error(s) occurred:

* aws_lambda_permission.events_endpoint_lambda_alb_permission: ResourceNotFoundException: Function not found: arn:aws:lambda:us-west-2:203867187697:function:dev_events_endpoint
	status code: 404, request id: 53419bff-1c2f-11e9-801b-1d7eac61b2be
```
### Expected Behavior

**aws_lambda_permission** should set up the correct Lambda permissions.

### Actual Behavior

**aws_lambda_permission** acts too fast and fails. 

### Important Factoids
In fact my lambda gets created by a module and then I throw lambda's ARN as an output. When I apply terraform rules the second time - everything goes well.

**Comment by davewongillies** [comment]
I'm seeing this too.

**Comment by bflad** [comment]
I have submitted a fix for this: #7327

**Comment by bflad** [comment]
The fix to allow the resource to automatically retry this error for up to 1 minute during resource creation has been merged and will release with version 1.57.0 of the Terraform AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.57.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1570-january-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
